final_task/rss_reader/validation_functions.py

```diff
@@ -0,0 +1,76 @@
+""" Module of validation functions.
+
+    Functions:
+    check_url(com_line_args, logger) -> True
+    check_internet_connection(com_line_args, logger) -> True
+    check_emptiness(news_collection, logger) -> True
+    check_version_arg(com_line_args, logger) -> None
+    check_limit_arg(news_collection, com_line_args, logger) -> limit (int) """
+
+import requests
+from urllib.request import Request, urlopen
+from urllib.error import URLError
+from exceptions import Error
+
+
+def check_url(com_line_args, logger):
+    """ Check URL function. """
+    try:
+        req = Request(com_line_args.source)
+        logger.info("Checking url.")
+        response = urlopen(req)
+    except ValueError:
+        logger.error("Invalid URL.")
+        raise Error("Please, check your URL.")
+    except URLError as e:
+        if hasattr(e, "reason"):
+            logger.error(f"Failed to reach a server. Reason: {e.reason}.")
+            raise Error("Please, check your URL.")
+        elif hasattr(e, 'code'):
+            logger.error("The server couldn\'t fulfill the request. "
+                         f"Error code: {e.code}")
+            raise Error("Service problem.")
+    else:
+        return True
+
+
+def check_internet_connection(com_line_args, logger):
+    """ Check internet connection function. """
+    try:
+        logger.info("Checking internet connection.")
+        response = requests.get("http://google.com", timeout=5)
+        return True
+    except requests.exceptions.ConnectionError:
+        logger.error("No internet connection.")
+        raise Error("Please, check your internet connection.")
```

Comment:
The function above does almost the same. Do you really need this one?

Reply:
I use two functions to tell what the problem is more precisely and to give more specific instructions for its correction. Do you think that it is unjustified to use an additional library for just one check?

Reply:
But you will not be able to use urlopen in the first function without internet connection anyway. 
Why not bundle it together if it does the same thing and does it in a similar way?

Reply:
I just don’t like the fact that the exception in urlopen both in the absence of the Internet and in the case of an incorrectly entered url is the same, albeit with different reasons. It is immediately clear that there is no Internet connection, in the case of request. But if necessary, I will fix it.)

Reply:
It is not that critical